tracer/src/Datadog.Trace/Util/DomainMetadata.cs

```diff
@@ -13,100 +12,80 @@ namespace Datadog.Trace.Util
     /// </summary>
     internal static class DomainMetadata
     {
-        private const string UnknownName = "unknown";
-        private static bool _initialized;
-        private static string _currentProcessName;
-        private static string _currentProcessMachineName;
-        private static int _currentProcessId;
-        private static bool _processDataUnavailable;
-        private static bool _domainDataPoisoned;
-        private static bool? _isAppInsightsAppDomain;
-
-        static DomainMetadata()
-        {
-            TrySetProcess();
-        }
+        private static readonly DomainMetadataImpl _instance = new DomainMetadataImpl();
 
-        public static string ProcessName
-        {
-            get
-            {
-                return !_processDataUnavailable ? _currentProcessName : UnknownName;
-            }
-        }
+        public static string ProcessName => _instance.ProcessName;
```

Comment:
When I mentioned having a singleton to avoid all the static field access, I meant something like this:

```
internal class DomainMetadata // not static
{
    public static readonly DomainMetadata Instance { get; } = new DomainMetadata();

    // same ctor, but private, the only instance that exists is DomainMetadata.Instance
    // same instance proprties
    // only static property is Instance
}
```

Then the API to use this is:
```
var Name = DomainMetadata.Instance.ProcessName;
```

The only reason I brought this up was to avoid multiple static field access in places like this that currently access 5 static fields (even after the changes in this PR)
https://github.com/DataDog/dd-trace-dotnet/blob/d009bf0e90018a34b5322a34abdba79228979cd9/tracer/src/Datadog.Trace/Logging/DatadogLogging.cs#L87-L89

If we change that to the following, there's only 1 static field access:
```
var domainMetadata = DomainMetadata.Instance; // reads a static field

loggerConfiguration.Enrich.WithProperty("MachineName", domainMetadata.MachineName);
loggerConfiguration.Enrich.WithProperty("Process", $"[{domainMetadata.ProcessId} {domainMetadata.ProcessName}]");
loggerConfiguration.Enrich.WithProperty("AppDomain", $"[{domainMetadata.AppDomainId} {domainMetadata.AppDomainName}]");
```

Reply:
But maybe it's not worth it if that the only case where that happens and that only runs once. Otherwise, we could just keep the static fields like we had them before and remove `DomainMetadataImpl`. `DomainMetadata` is still much cleaner than it was before even without that 😊

Reply:
you're right!, I made the change in 14b604c

Reply:
Thanks!